ckan/lib/helpers.py

```diff
@@ -1687,17 +1692,36 @@ def time_ago_from_timestamp(timestamp: int) -> str:
     return formatters.localised_nice_date(datetime_, show_date=False)
 
 
+def _object_display_name(obj_dict: dict[str, Any]) -> str:
+    return get_translated(obj_dict, 'title') or obj_dict['name']
+
+
 @core_helper
 def dataset_display_name(
         package_or_package_dict: Union[dict[str, Any], model.Package]) -> str:
+    if not isinstance(package_or_package_dict, dict):
+        package_or_package_dict = logic.get_action(u'package_show')(
+            {}, {u'id': package_or_package_dict.id})
+    return _object_display_name(package_or_package_dict)  # type: ignore
 
-    if isinstance(package_or_package_dict, dict):
-        return get_translated(package_or_package_dict, 'title') or \
-            package_or_package_dict['name']
-    else:
-        # FIXME: we probably shouldn't use the same functions for
-        # package dicts and real package objects
-        return package_or_package_dict.title or package_or_package_dict.name
+
+@core_helper
+def group_display_name(
```

Comment:
If we're making a new helper can we not follow the one above that works on models or dicts and only accept dicts?

Reply:
New helper is only doing dicts now. Simplified a lot, which is nice.